hooks/pre/13-dns_check.rb

```diff
@@ -0,0 +1,21 @@
+# Check to verify forward/reverse dns matches hostname
+
+if app_value(:disable_system_checks)
+  logger.warn 'Skipping system checks.'
+else
+  hostname = `hostname -f`
+  ip = `host #{hostname} | awk '{print $4}'`
```

Comment:
I've been told using ```host``` is considered bad practice because it's unpredictable. Please use ```getent hosts #{hostname} | awk '/STREAM { print $1 }'``` to the get IP address using the exact same logic as the OS. Note this can be IPv6 but I think that's good here since that is what the OS would try to connect to anyway. If you want to explicitly test all IPs (which is even better) you can use ```getent ahosts #{hostname} | awk '/STREAM/ { print $1 }'``` to get both IPv4 and IPv6.

Reply:
Hello, wouldn't it be better to use Sockets and rubydns, rather than calling directly system commands which may change output at any particular time?

Reply:
Given getent is part of glibc I'm not that worried about it suddenly changing but pure ruby code can also work just as well.

Reply:
@GAZDOWN were not using sockets, this method works fine.